nemo_rl/algorithms/grpo.py

```diff
@@ -604,13 +605,33 @@ def grpo_train(
                     )
 
             with timer.time("data_processing"):
+                use_overlong_filtering = master_config["grpo"].get(
+                    "overlong_filtering", False
+                )
+                if use_overlong_filtering and "truncated" in repeated_batch:
+                    loss_multiplier = repeated_batch["loss_multiplier"].clone()
+                    truncated = repeated_batch["truncated"]
+
+                    if isinstance(truncated, list):
+                        truncated = torch.tensor(truncated, dtype=torch.bool)
+
+                    loss_multiplier[truncated] = 0
+                    repeated_batch["loss_multiplier"] = loss_multiplier
                 # Add loss mask and advantages to each message in LLMMessageLogType
                 for i, message_log in enumerate(repeated_batch["message_log"]):
+                    overlong_filtered = (
+                        use_overlong_filtering and repeated_batch["truncated"][i]
+                    )
                     for j, message in enumerate(message_log):
                         if message["role"] == "assistant":
-                            message["token_loss_mask"] = torch.ones_like(
-                                message["token_ids"]
-                            )
+                            if overlong_filtered:
```

Comment:
this part of the logic is not required. Just setting the loss_multiplier is sufficient (corresponds to sample_mask) in the loss function